### PR TITLE
Fiixed issues in dashboard designer and dashboard banner feature

### PR DESCRIPTION
**Pull request by lasanthaS**

### CODE REVIEW

File: apps/portal/controllers/apis/banners.jag [hunk 1]
```diff
@@ -56,21 +57,30 @@
...
+                    
+                    // decode the base64 string
+                    var decodedData = Packages.javax.xml.bind.DatatypeConverter.parseBase64Binary(encodedData);
+
+                    // save the buffer in the registry
+                    dashboards.saveBanner(dashboardId, username, 'banner', 'image/jpg', decodedData); 
+
+                } else {
+                    
+					// when a viewer removes the original banner which is available globally, a empty resource is saved as
```
Comment (TanyaM) [inline apps/portal/controllers/apis/banners.jag]: Check indentation

File: apps/portal/controllers/apis/banners.jag [hunk 2]
```diff
@@ -56,21 +57,30 @@
...
+                } else {
+                    
+					// when a viewer removes the original banner which is available globally, a empty resource is saved as 
+					// the personalized banner in order to suppress the original banner
+                    dashboards.saveBanner(dashboardId, username, 'banner', 'image/jpg', ''); 
+                
+                }
                 
             } catch(e) {
                 log.error('Error occurred while saving the banner');
```
Comment (TanyaM) [inline apps/portal/controllers/apis/banners.jag]: log e as well

File: apps/portal/controllers/apis/banners.jag [hunk 3]
```diff
@@ -82,6 +92,11 @@
         // delete banners
         if (method === 'DELETE') {
             dashboards.deleteBanner(dashboardId, username);
+            
+			// return the status of the banner (as JSON data)
+            var banner = dashboards.getBanner(dashboardId, username);
+			var bannerStatus = { globalBannerExists: banner.globalBannerExists, customBannerExists: banner.customBannerExists };
```
Comment (TanyaM) [inline apps/portal/controllers/apis/banners.jag]: Check indentation

File: apps/portal/js/designer.js [hunk 4]
```diff
@@ -2552,20 +2562,49 @@ $(function () {
...
+                });
+                
+            } else {
+                // remove the banner 
+                $.ajax({
+                    url: $form.attr('action'),
+                    type: 'DELETE',
+                    dataType: 'json'
+                }).success(function (d) {
+
```
Comment (TanyaM) [inline apps/portal/js/designer.js]: New line

File: apps/portal/controllers/apis/banners.jag (hunk 1 above)
Comment (lasanthaS) [inline apps/portal/controllers/apis/banners.jag]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/174

File: apps/portal/controllers/apis/banners.jag (hunk 2 above)
Comment (lasanthaS) [inline apps/portal/controllers/apis/banners.jag]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/174

File: apps/portal/controllers/apis/banners.jag (hunk 3 above)
Comment (lasanthaS) [inline apps/portal/controllers/apis/banners.jag]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/174

File: apps/portal/js/designer.js (hunk 4 above)
Comment (lasanthaS) [inline apps/portal/js/designer.js]: Fixed in the PR https://github.com/wso2/carbon-dashboards/pull/174

**Pull request merged**